commit 0574d3ee4a6d829cee2bbc2595802380d73f68a5
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2019-06-24T14:35:36-07:00 (4 months ago)
    
    TOOLS-2280 gerrit operator guide should have note about git gc

diff --git a/docs/operator/README.md b/docs/operator/README.md
index 790c785..26e6676 100644
--- a/docs/operator/README.md
+++ b/docs/operator/README.md
@@ -146,6 +146,32 @@ to get the settings right, and we likely won't discover if they're wrong until
 something bad has already happened.)
 
 
+### Using `git gc` on `illumos-joyent`
+
+Gerrit git repositories need to be regularly garbage-collected, as the review
+process often results in quite a lot of dangling commits and sub-optimal pack
+structures. Gerrit itself does not do any automatic periodic garbage collection,
+and today we run this by hand on an as-needed basis.
+
+While most repositories in Gerrit currently can be garbage collected by running
+`gerrit gc` via the SSH interface, the `illumos-joyent` repo is too large for
+this tool to run. If you start a `gerrit gc` on `illumos-joyent` it will run
+for around 6-12 hours while making the Gerrit server very slow and is likely to
+eventually crash it.
+
+In accordance with the Gerrit documentation, you can run `git gc` by hand in
+the repository directly for cases like this. This is best done by logging into
+the Gerrit zone using `docker exec` and going to the directory
+`/var/gerrit/review_site/git/joyent/illumos-joyent.git`, where you can run
+`su -l -s /bin/bash gerrit2` and then `git gc --aggressive`.
+
+It's important to make sure to use the `su` command to change to the `gerrit2`
+user, or else you will mess up permissions on parts of the git repo on disk (if
+you do so the first thing you'll likely notice is that pushes of new reviews
+stop working -- reading existing reviews is likely to still work, so don't
+rely on just doing that to test if you're unsure whether you messed it up).
+
+
 ## Deployment notes
 
 cr.joyent.us is deployed in us-west-1 (behind TLS), using CNS for internal and
